### ISSUE TITLE
Unable to download the youtube video KkztNmlYMSQ although I can view it in firefox

### ISSUE THREAD

**Issue by HansEhrbar** [rg3/youtube-dl#6475]
Unable to download the youtube video KkztNmlYMSQ
I have Ubuntu LTS and just wgot the latest youtube-dl which has the same error as the ubuntu youtube dl

he5513@x5-601000z7:~/Downloads/degrowth$ /usr/local/bin/youtube-dl -v KkztNmlYMSQ
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'KkztNmlYMSQ']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.07.28
[debug] Python version 2.7.6 - Linux-3.13.0-61-generic-x86_64-with-Ubuntu-14.04-trusty
[debug] exe versions: avconv 9.18-6, avprobe 9.18-6, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] KkztNmlYMSQ: Downloading webpage
[youtube] KkztNmlYMSQ: Downloading video info webpage
[youtube] KkztNmlYMSQ: Extracting video information
[youtube] KkztNmlYMSQ: Downloading DASH manifest
[youtube] KkztNmlYMSQ: Downloading DASH manifest
WARNING: Your copy of avconv is outdated and unable to properly mux separate video and audio files, youtube-dl will download single file media. Update avconv to version 10-0 or newer to fix this.
[debug] Invoking downloader on u'https://r16---sn-nx57yn7y.googlevideo.com/videoplayback?key=yt5&ip=166.70.24.4&ms=au&mt=1438839750&mv=m&id=o-AGWzSns0RAlXDyQsg6Hzlbd4s4vy-H6T900_b5HGwqzj&mm=31&mn=sn-nx57yn7y&upn=En_ipFBBLKw&lmt=1410480132578736&requiressl=yes&dur=7213.186&pl=20&itag=22&source=youtube&expire=1438861477&mime=video%2Fmp4&signature=706CAB28C2649E2ECB82C40C40559192EF4CD775.2E3311C1B56025A28800BCC77BD7862A86457E82&ratebypass=yes&sver=3&ipbits=0&fexp=9408512%2C9408710%2C9409172%2C9412515%2C9412913%2C9414764%2C9415357%2C9415365%2C9415485%2C9416126%2C9416179%2C9416778%2C9417346%2C9417703%2C9417721%2C9418153%2C9418204%2C9418823&sparams=dur%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cpl%2Cratebypass%2Crequiressl%2Csource%2Cupn%2Cexpire&initcwndbps=1982500'
[download] Resuming download at byte 2097136
[download] Got server HTTP error. Retrying (attempt 1 of 10)...
[download] Got server HTTP error. Retrying (attempt 2 of 10)...
[download] Got server HTTP error. Retrying (attempt 3 of 10)...
[download] Got server HTTP error. Retrying (attempt 4 of 10)...
[download] Got server HTTP error. Retrying (attempt 5 of 10)...
[download] Got server HTTP error. Retrying (attempt 6 of 10)...
[download] Got server HTTP error. Retrying (attempt 7 of 10)...
[download] Got server HTTP error. Retrying (attempt 8 of 10)...
[download] Got server HTTP error. Retrying (attempt 9 of 10)...
[download] Got server HTTP error. Retrying (attempt 10 of 10)...
ERROR: giving up after 10 retries
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/http.py", line 59, in real_download
    data = self.ydl.urlopen(request)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1732, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.7/urllib2.py", line 410, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 523, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 448, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 531, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
HTTPError: HTTP Error 500: Internal Server Error

**Comment by remitamine** [comment]
`HTTPError: HTTP Error 500: Internal Server Error` it's an error from the server.
now, i can download the same format you get(22):
```
youtube-dl -v KkztNmlYMSQ
[debug] System config: []
[debug] User config: ['--external-downloader', 'aria2c', '--external-downloader-args', '-x 16 --check-certificate=false', '-f', '[height=720]', '--sub-lang', 'ar,en', '--write-sub']
[debug] Command-line args: ['-v', 'KkztNmlYMSQ']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.07.28
[debug] Python version 3.4.3 - Linux-4.1.4-1-ARCH-x86_64-with-arch-Arch-Linux
[debug] exe versions: ffmpeg 2.7.2, ffprobe 2.7.2, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] KkztNmlYMSQ: Downloading webpage
[youtube] KkztNmlYMSQ: Downloading video info webpage
[youtube] KkztNmlYMSQ: Extracting video information
WARNING: video doesn't have subtitles
[youtube] KkztNmlYMSQ: Downloading DASH manifest
[youtube] KkztNmlYMSQ: Downloading DASH manifest
[debug] Invoking downloader on 'https://r16---sn-hpa7lne7.googlevideo.com/videoplayback?id=o-AAWkFTAk1K1kQsrTvlpKXhQpMRXeKVj2BbVMVowgkOHM&signature=83F5F2F141C518B3D23EB698BAAE65C14838061C.B763F99F7707FE1E647C928C442A8E5CD493A55D&ms=au&mv=m&upn=zSyKpmFoygU&mt=1438855109&expire=1438876812&pl=17&mn=sn-hpa7lne7&ip=197.205.120.109&mm=31&requiressl=yes&initcwndbps=356250&source=youtube&ipbits=0&lmt=1410480132578736&sparams=dur%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cpl%2Cratebypass%2Crequiressl%2Csource%2Cupn%2Cexpire&fexp=9405185%2C9405969%2C9408710%2C9410705%2C9412774%2C9415365%2C9415485%2C9415820%2C9416126%2C9416326%2C9416687%2C9416729%2C9416778%2C9416898%2C9418153%2C9418392%2C9418780&dur=7213.186&ratebypass=yes&sver=3&itag=22&key=yt5&mime=video%2Fmp4'
[download] Destination: Podium - A worldwide Energy Revolution - Degrowth Leipzig 2014-KkztNmlYMSQ.mp4
[debug] aria2c command line: aria2c -c -x 16 --check-certificate=false --out 'Podium - A worldwide Energy Revolution - Degrowth Leipzig 2014-KkztNmlYMSQ.mp4.part' --header 'Accept-Encoding: gzip, deflate' --header 'User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/20.0 (Chrome)' --header 'Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7' --header 'Accept-Language: en-us,en;q=0.5' --header 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' -- 'https://r16---sn-hpa7lne7.googlevideo.com/videoplayback?id=o-AAWkFTAk1K1kQsrTvlpKXhQpMRXeKVj2BbVMVowgkOHM&signature=83F5F2F141C518B3D23EB698BAAE65C14838061C.B763F99F7707FE1E647C928C442A8E5CD493A55D&ms=au&mv=m&upn=zSyKpmFoygU&mt=1438855109&expire=1438876812&pl=17&mn=sn-hpa7lne7&ip=197.205.120.109&mm=31&requiressl=yes&initcwndbps=356250&source=youtube&ipbits=0&lmt=1410480132578736&sparams=dur%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cpl%2Cratebypass%2Crequiressl%2Csource%2Cupn%2Cexpire&fexp=9405185%2C9405969%2C9408710%2C9410705%2C9412774%2C9415365%2C9415485%2C9415820%2C9416126%2C9416326%2C9416687%2C9416729%2C9416778%2C9416898%2C9418153%2C9418392%2C9418780&dur=7213.186&ratebypass=yes&sver=3&itag=22&key=yt5&mime=video%2Fmp4'
[#def419 10MiB/1.4GiB(0%) CN:16 DL:237KiB ETA:1h47m37s]
```

**Comment by synhaptein** [comment]
I have the same issue with https://www.youtube.com/watch?v=VObwmJXUdCY

```
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'VObwmJXUdCY']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.08.16
[debug] Python version 2.7.10 - Darwin-13.4.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 2.7.2, ffprobe 2.7.2, rtmpdump 2.4
[debug] Proxy map: {}
```

**Comment by HansEhrbar** [comment]
I just tried https://www.youtube.com/watch?v=VObwmJXUdCY and it worked for me.  Hans
________________________________

**Comment by synhaptein** [comment]
Related issue? #6598 It still fails for me...

**Comment by sirkkalap** [comment]
I just reproduced @synhaptein 's reported issue. I get "ERROR: unable to download video data: HTTP Error 302: The HTTP server returned a redirect error that would lead to an infinite loop."

- https://gist.github.com/sirkkalap/0da12cd4742e72b6a778

**Comment by sirkkalap** [comment]
@synhaptein

After running `youtube-dl --rm-cache-dir` the result changed a bit and youtube-dl now reports **server** error 500. Maybe there is some issue with the source in some formats?

- https://gist.github.com/sirkkalap/1e7597da2463d93e6556

**Comment by synhaptein** [comment]
I have no idea, because it works with a browser directly on youtube. I tried different format (video only) and could not read the resulting video file.

**Comment by sirkkalap** [comment]
If I am not mistaken, the browser might use a different format for both video and audio. Maybe it is the audio that has some missing or malfunctional format, that the ytdl tries to get. I am sorry, but I am just guessing here, since I really do not know all the internals. But I was able to reproduce your error in two different ways :).

**Comment by sirkkalap** [comment]
And I was also able to download the clip with Safari (Click to Flash) HTML5 replacement player. The format I got this way was H.264 1280x760 AAC 44100 kHz Stereo. File size 22 224 015 bytes.

**Comment by yan12125** [comment]
All examples mentioned here are working now. All of them use DASH segmented format. DASH downloader is implemented in #5886 and getting more mature later. They should work fine in the latest version.

Closing. Please go to #3721 if the problem is still.
